executor/builder.go

```diff
@@ -226,9 +226,14 @@ func (b *executorBuilder) buildTableScan(v *plan.TableScan) Executor {
 		}
 		if len(remained) == 0 {
 			e.allFiltersPushed = true
-			return e
 		}
-		return b.buildFilter(e, remained)
+		var ex Executor
+		if txn.IsReadOnly() {
+			ex = e
+		} else {
+			ex = b.buildUnionScanExec(e)
```

Comment:
If the table is dirty, do not push aggregate functions, table scan can still use XAPI.